app/interactors/add_movie_to_round.rb

```diff
@@ -3,26 +3,42 @@
 class AddMovieToRound
   include Hanami::Interactor
 
-  def initialize(import_service: ImportTmdbMovie.new)
-    @import_service = import_service
+  expose :movie, :round
+
+  def initialize(import_tmdb_movie_interactor: ImportTmdbMovie.new)
+    @import_tmdb_movie_interactor = import_tmdb_movie_interactor
   end
 
-  def call(round:, tmdb_id:)
+  def call(round:, tmdb_movie_id:)
     @round = round
-    @tmdb_id = tmdb_id
-
-    movie = Movie.find_by(tmdb_id: @tmdb_id)
-    movie = imported_movie if movie.blank?
+    @tmdb_movie_id = tmdb_movie_id
 
-    movie.errors.full_messages.each { |msg| error(msg) } unless @round.update(movie: movie)
+    fetch_movie
```

Comment:
Here I'm embracing the deliberate imperative style (more than imperative, I think of these functions as _commands_) that is recommended by Hanami **for these objects**.

Notice that I don't think that local logic should use mutation or be imperative per se, of course. But since we are causing so many side effects -- fetching from the DB or an external API (in which case we do a record creation on the DB first --, then updating a value and persisting it **or** failing with errors, I think using accessors with memoization is a terrible thing that makes this code "very much not explicit".

This is, of course, debatable.